lib/layer.js

```diff
@@ -173,3 +173,15 @@ function decode_param(val){
     throw err
   }
 }
+
+function once(fn) {
+  var called = false
+
+  return function (err) {
+    if (!err) {
+      if (called) { throw new Error('next() cannot be called twice') }
```

Comment:
Please use new lines for the block, which helps a lot for line code coverage metrics and easier to read.